crates/holochain/src/conductor/tests/agent_info.rs

```diff
@@ -0,0 +1,328 @@
+use holochain_wasm_test_utils::TestWasm;
+use kitsune2_api::AgentInfoSigned;
+use kitsune2_core::Ed25519Verifier;
+
+// in these tests we set up a mix of apps and including clone cells so we can test
+// different varieties of combinations in the app_agent_info case, and we use the same setup in the admin_agent_info
+// for parity.
+async fn setup_tests() -> (
+    DnaHash,
+    DnaHash,
+    DnaHash,
+    String,
+    String,
+    ClonedCell,
+    SweetConductor,
+) {
+    // Create three different DNAs
+    let dna1 = SweetDnaFile::unique_from_test_wasms(vec![TestWasm::Create]).await;
+    let dna2 = SweetDnaFile::unique_from_test_wasms(vec![TestWasm::Create]).await;
+    let dna3 = SweetDnaFile::unique_from_test_wasms(vec![TestWasm::Create]).await;
+
+    // Install two different apps on a conductor: app1 (dna1, dna2) and app2 (dna3)
+    let config = SweetConductorConfig::standard();
+    let mut conductor = SweetConductor::from_config(config.clone()).await;
+
+    let app1_id: InstalledAppId = "app1".into();
+    let app2_id: InstalledAppId = "app2".into();
+
+    let installed_app1_id = conductor
+        .setup_app(&app1_id, &[dna1.0.clone(), dna2.0.clone()])
+        .await
+        .unwrap()
+        .installed_app_id()
+        .clone();
+
+    // Install app2 on conductors 1 and 2
+    let installed_app2_id = conductor
+        .setup_app(&app2_id, &[dna3.0.clone()])
+        .await
+        .unwrap()
+        .installed_app_id()
+        .clone();
+
+    // Create a disabled clone cell for app1 on conductor[0]
+    let clone_cell_r = conductor
+        .create_clone_cell(
+            &installed_app1_id,
+            CreateCloneCellPayload {
+                role_name: dna1.0.dna_hash().to_string(),
+                modifiers: DnaModifiersOpt::none().with_network_seed("test_seed".to_string()),
+                membrane_proof: None,
+                name: Some("disabled_clone".into()),
+            },
+        )
+        .await;
+    let clone_cell = clone_cell_r.unwrap();
+
+    (
+        dna1.0.dna_hash().clone(),
+        dna2.0.dna_hash().clone(),
+        dna3.0.dna_hash().clone(),
+        installed_app1_id,
+        installed_app2_id,
+        clone_cell,
+        conductor,
+    )
+}
+
+fn make_agent(space: kitsune2_api::SpaceId) -> String {
+    let local = kitsune2_core::Ed25519LocalAgent::default();
+    let created_at = kitsune2_api::Timestamp::now();
+    let expires_at = created_at + std::time::Duration::from_secs(60 * 20);
+    let info = kitsune2_api::AgentInfo {
+        agent: kitsune2_api::LocalAgent::agent(&local).clone(),
+        space,
+        created_at,
+        expires_at,
+        is_tombstone: false,
+        url: None,
+        storage_arc: kitsune2_api::DhtArc::FULL,
+    };
+    let info =
+        futures::executor::block_on(kitsune2_api::AgentInfoSigned::sign(&local, info)).unwrap();
+    info.encode().unwrap()
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn admin_agent_info() {
```

Comment:
I appreciate the test! There's another websocket test https://github.com/holochain/holochain/blob/develop/crates/holochain/src/conductor/interface/websocket.rs#L1224

I think that can be deleted and the one you wrote be kept.

Reply:
done